### DISCUSSION TITLE
How to block all unencypted traffic coming on any local port ?

### DISCUSSION THREAD

**Discussion by MOHITCHITLANGE** [strongswan/strongswan#692] [Q&A]
I was looking to block all the unencrypted packets coming on my host to make it completely secure.
I saw the way set the trap but that allows to block all the traffic coming from specified remote port and not on local port.
Can someone help me here.

**Comment by tobiasbrunner** [comment]
> and not on local port

Why not?

Block policies and firewall rules can also be used to block unencrypted traffic. Also see [SecurityRecommendations](https://wiki.strongswan.org/projects/strongswan/wiki/SecurityRecommendations#Tunnel-Shunting) for some hints.

**Reply by MOHITCHITLANGE** [reply]
connections {
        conn-host {
            local_addrs = %any
            remote_addrs = %any
            local {
                auth = psk
            }
            remote {

                auth = psk
            }
            children {
                specv-net {
                    esp_proposals = aes128-sha256-ecp256,aes256-sha256-ecp256,aes128gcm16-ecp256,aes256gcm16-ecp256
                    local_ts  = 192.168.0.0/16
                    remote_ts = 192.168.0.0/16
                    start_action = trap
                }
                drop-unwanted {
                    local_ts  = 192.168.0.0/16
                    remote_ts  = 192.168.0.0/16
                    mode = drop
                    priority = 200000
                    start_action = trap
                }
            }
        }
    }

**Reply by MOHITCHITLANGE** [reply]
Above is my swanctl.conf, I am able to create tunnel from other side but unable to ping or communicate on the IP's after creating tunnel as well.

**Reply by tobiasbrunner** [reply]
If you hadn't configured the same subnets on both ends, you'd have to enable `policies_fwd_out` for the `specv-net` child. But since the subnets are the same, the installed _fwd_ policy for the inbound direction should already be enough. But there could be some routing issues (exactly because you configured the same subnets on both sides) or a firewall problem. Also, there is no need to set the priority manually, the default priority for drop policies is always lower than that of regular or trap policies.

**Reply by MOHITCHITLANGE** [reply]
Got it. Thanks.
